### DISCUSSION TITLE
DAO DAO <> Akash

### DISCUSSION THREAD

**Discussion by elsehow** [akash-network/community#570] [General]
This is an open-ended discussion about the potential upsides to deploying [DAO DAO](https://daodao.zone/) on Akash.

I understand Akash has some pain points around vesting, particularly on community spend props.

This is something DAO DAO can easily provide to Akash today. We provide this functionality to several ecosystem chains (Atom Accelerator, for example). 

I’ve also heard discussions in sig-economics about potential use cases for ICA. DAO DAO can provide a [polytone](https://github.com/DA0-DA0/polytone) deployment—our fuller-featured alternative to interchain accounts (ICA)—built entirely with CosmWasm. 

Of course, a full DAO DAO deployment can also provide DAO DAO DAOs on Akash, which would be convenient for teams that deploy software together. Enabling fully on-chain decentralized organizations to manage servers and deploy software is something the world has never seen—tantamount to a fully web3 software company—and we can do it right now.

Deploying these features would require CosmWasm on Akash. CosmWasm is relatively battle-tested, and if contract deployments are permissioned, we can assure that only audited contracts are deployed. (DAO DAO’s contracts currently protect $150M across various ecosystem chains).  

More about DAO DAO below.

## About DAO DAO

With DAO DAO, Akash projects and communities would have immediate access to best-in-class DAO tooling without needing to build their own governance modules, smart contracts, UI, and more. Try it today:[ https://daodao.zone](https://daodao.zone/)
Our mission is to build better, more flexible governance systems. Getting governance right is crucial to the success of decentralized communities, protocols, and entire blockchains, and DAO DAO makes it as easy as clicking a few buttons to experiment with and scale governance infrastructure.

Today, over $150M is managed by 5,026 DAO DAO DAOs across 8 chains (and growing). Our smart contracts and UI have been battle tested by 10,687 proposals encompassing 31,684 unique DAO members for over 2 years. Over $1M (and many, many hours) have been put into the necessary research and development, as well as the perfection of each feature, with a solid chunk devoted to numerous security audits.

The biggest features we can bring to Akash:

- NFT DAOs, Token DAOs, and Membership (multisig-like) DAOs
- Beautiful UI for making proposals, voting, and exploring DAOs
- Cross-Chain Accounts via Polytone
- Vesting and Retroactive Payment solutions
- Sophisticated organizational structures using SubDAOs and Authz

And much more:

- Treasury spends and swaps
- Staking management
- Veto and approval flows
- NFT creation and management
- On-chain, governance gated key / value store
- Proposal inbox
- Customizable DAO widgets
- DAO Press: publishing tools for a DAO-run blog, official communications, and more
- Bulk actions import
-Smart contract management (instantiate, execute, migrate, update admin)
- DAO-run validators (check out ours here:[ https://daodao.zone/dao/juno185hgkqs8q8ysnc8cvkgd8j2knnq2m0ah6ae73gntv9ampgwpmrxqc5vwdr](https://daodao.zone/dao/juno185hgkqs8q8ysnc8cvkgd8j2knnq2m0ah6ae73gntv9ampgwpmrxqc5vwdr))
- Push and email notifications
- Native mobile support
- and more...

Our immediate roadmap includes:
- Staking rewards
- Voting with locked vesting tokens
- Voting delegations
- Treasury rebalancing management
- Polytone Interchain Accounts and Queries

In addition to DAO DAO contracts, Akash will receive [Polytone](https://github.com/DA0-DA0/polytone) to allow for accounts, smart contracts, or DAOs on Akash to control accounts on other chains, and vice versa.

We are excited for the new possibilities that will be unlocked for developers and the wider Akash community. For example, a smart contract (or DAO) on another CosmWasm chain can control an account on Akash to deploy and manage servers. Akash teams can also use it to manage outposts across the Interchain. Moreover, Polytone includes an easy to use Interchain Queries implementation, allowing for smart contracts on other chains to reliably query information from Akash.

## Audits
- [DAO DAO v1](https://github.com/oak-security/audit-reports/blob/master/DAO%20DAO/2022-06-22%20Audit%20Report%20-%20DAO%20DAO%20v1.0.pdf)
- [DAO DAO v2](https://github.com/oak-security/audit-reports/blob/master/DAO%20DAO/2023-02-06%20Audit%20Report%20-%20DAO%20DAO%202%20v1.0.pdf)
- [Polytone (smart contract interchain accounts)](https://github.com/oak-security/audit-reports/blob/master/Polytone/2023-06-05%20Audit%20Report%20-%20Polytone%20v1.0.pdf)
- [Vesting and Payroll Factory](https://github.com/oak-security/audit-reports/blob/master/DAO%20DAO/2023-03-22%20Audit%20Report%20-%20DAO%20DAO%20Vesting%20and%20Payroll%20Factory%20v1.0.pdf)
- [Token Factory and NFT upgrades](https://github.com/oak-security/audit-reports/blob/master/DAO%20DAO/2023-10-16%20Audit%20Report%20-%20DAO%20DAO%20Updates%20v1.0.pdf)
- [Veto upgrade](https://github.com/oak-security/audit-reports/blob/master/DAO%20DAO/2024-01-10%20Audit%20Report%20-%20DAO%20DAO%20Veto%20v1.0.pdf)

## Codebases
- DAO Contracts:[ https://github.com/DA0-DA0/dao-contracts](https://github.com/DA0-DA0/dao-contracts)
- DAO DAO UI:[ https://github.com/DA0-DA0/dao-dao-ui](https://github.com/DA0-DA0/dao-dao-ui)
- Polytone:[ https://github.com/DA0-DA0/polytone](https://github.com/DA0-DA0/polytone)

**Comment by elix1er** [comment]
bump

**Comment by poroburu** [comment]
I was about to make an issue on Polytone suggesting that Polytone create a Cosmos SDK v47 Go module for the Polytone protocol. Specifically for Akash.

DAOs provisioning cloud resources on Akash would be a significant addressable market for both DAO DAO and Akash. Cross Chain Accounts on DAO DAO are simply great UX to use.

https://github.com/akash-network/support/issues/212
https://github.com/akash-network/support/issues/171

https://github.com/DA0-DA0/polytone/issues/68

**Reply by noahsaso** [reply]
> I was about to make an issue on Polytone suggesting that Polytone create a Cosmos SDK v47 Go module for the Polytone protocol. Specifically for Akash.
> 
> DAOs provisioning cloud resources on Akash would be a significant addressable market for both DAO DAO and Akash. Cross Chain Accounts on DAO DAO are simply great UX to use.
> 
> [akash-network/support#212](https://github.com/akash-network/support/issues/212) [akash-network/support#171](https://github.com/akash-network/support/issues/171)
> 
> [DA0-DA0/polytone#68](https://github.com/DA0-DA0/polytone/issues/68)

Posted a reply over in the Polytone issue: https://github.com/DA0-DA0/polytone/issues/68#issuecomment-2145966243

**Comment by 31trainman** [comment]
Are you working with trigs_0 on this?

**Reply by Trigs-0** [reply]
Sure am!

**Comment by dominikusbrian** [comment]
This is very very exciting. I currently is researching and testing various infrastructure to build what I called RnD-DAOs, and ideally I want it to live natively in a cross-chain environment, and making Akash sort of the Home Environment (because I want and expect this RnD-DAOs to have various activities that utilies Akash Supercloud). The members for thid RnD-DAOs going to be both Humans but also AI-Agents.   

Since Akash currently doesn't have native Smart Contract infra, building from ground up will be challenging, that being said, 
the proposed:
The biggest features we can bring to Akash:

> NFT DAOs, Token DAOs, and Membership (multisig-like) DAOs
> Beautiful UI for making proposals, voting, and exploring DAOs
> Cross-Chain Accounts via Polytone
> Vesting and Retroactive Payment solutions
> Sophisticated organizational structures using SubDAOs and Authz

Sounds really promising as a starting point to build the infrastructure integration in Akash. The CosmWasm powered Polytone, also looks like a very ideal way for me to build "Bank Account" for my AI Agents so they can perform various economic activities (do job and get paid, ask for compute/data resouce and pay things accordingly). 

I will go doing deep-dive on Polytone and other DAO DAOs infrastructure and toolkits, and will be paying great attention to this thread. 
I am both a compute resource provider and daily deployment user in Akash ecosystem, so I would be happy to provide direct assistance to this initiative in the way I can. I am a computational material scientist by professional training, but has been working with HPC (High Performance Computing) and ML/AI for the past 5 years or so and have enough technical skill to implement and build related stuff. In terms of smart contract development, in Rust or Go, I am considerably new and just started learning and developing for the past 1-1.5 years, still much to learn.  

Looking forward to learn and possibly have further discussion on upcoming roadmap and who are working on what in terms of moving this discussion forward and toward becoming a solid proposal with actionable items. 

Best, 
Domi.
 

ps: Thanks to Tyler Wright from Akash for bringing this thread to my attention when we were having group discussion about NFT and Smart Contract on Akash.
